web/src/emoji_picker.js

```diff
@@ -320,6 +320,32 @@ function process_enter_while_filtering(e) {
             }
         }
     }
+
+    const $searchInput = $("#emoji-search-query");
+    const $clearButton = $("#emoji_search_clear");
+    $(document).on("input", "#emoji-search-query", () => {
+        if ($searchInput.val() === "") {
+            $clearButton.css("display", "none");
+        } else {
+            $clearButton.css("display", "block");
+        }
+    });
+
+    $(document).on("click", "#emoji_search_clear", async (e) => {
+        e.stopPropagation();
+        $("#emoji-search-query").val("");
+        show_emoji_catalog();
+        $clearButton.css("display", "none");
+    });
+    $(document).on("keydown", "#emoji_search_clear", async (e) => {
+        e.stopPropagation();
+        const id = e.key || e.which || 0;
+        if (id === 13) {
+            $("#emoji-search-query").val("");
+            show_emoji_catalog();
+            $clearButton.css("display", "none");
+        }
+    });
```

Comment:
These don't belong inside `process_enter_while_filtering`. Let's place them with other event handlers.

Reply:
Also, we prefer snake_case to camelCase for variable names.